### PR TITLE
add zuul configuration

### PR DESCRIPTION
**Pull request by pc-m**

### CODE REVIEW

**Comment by pc-m** [comment]
recheck

File: tox.ini [hunk 1]
```diff
@@ -35,3 +35,15 @@ deps =
...
+
+[testenv:linters]
+skip_install = true
+basepython = python3
+deps = flake8
+       flake8-colors
+       black
+commands =
+    # do not fail for black
+    -black --skip-string-normalization --check .
```
Comment (sileht) [inline tox.ini]: Looks like it passes black, so you can remove the `-`

**Comment by sileht** [comment]
wrong btn

**Comment by pc-m** [comment]
This branch depends on https://github.com/wazo-pbx/sf-config/pull/8 or enabling sudo to be able to install `bindep.txt` on the container

File: tox.ini (hunk 1 above)
Comment (pc-m) [reply tox.ini]: I get this when running black
```
All done! 💥 💔 💥
176 files would be reformatted, 101 files would be left unchanged.
ERROR: InvocationError for command '/home/pcm/dev/wazo/wazo-dird/.tox/linters/bin/black --skip-string-normalization --check .' (exited with code 1)
___________________________________________________________________________________________________________________ summary ___________________________________________________________________________________________________________________
ERROR:   linters: commands failed

```

**Comment by pc-m** [comment]
recheck

**Comment by pc-m** [comment]
recheck

File: tox.ini (hunk 1 above)
Comment (sileht) [reply tox.ini]: Oh ok, so just run 'tox -eblack', commit all changed files, and the you can remove it :)

**Comment by sileht** [comment]
recheck

File: tox.ini [hunk 2]
```diff
@@ -35,3 +36,35 @@ deps =
...
+    flake8
+
+[testenv:integration]
+usedevelop = true
+deps = -rintegration_tests/test-requirements-for-tox.txt
```
Comment (sileht) [inline tox.ini]: should be:
```
-rintegration_tests/test-requirements.txt
```

File: tox.ini [hunk 3]
```diff
@@ -35,3 +36,35 @@ deps =
...
+    pytest {posargs}
+whitelist_externals:
+    make
+
+[flake8]
+exclude =
+    .tox
+    .eggs
+    integration_tests/assets/scripts
+show-source = true
```
Comment (sileht) [inline tox.ini]: You should add at least those ignore to make flake8 compatible with black:
```
# W503: line break before binary operator
# E203: whitespace before ':' warnings
ignore = E203, W503
```

Comment (fblackburn1) [reply tox.ini]: why E203?

Comment (sileht) [reply tox.ini]: black adds a space before the `:` while flake8 complains you should remove it: https://github.com/python/black/issues/315

Comment (pc-m) [reply tox.ini]:
```python
return backends[offset : limit + offset]
```
This is an example where I've seen add a space before a `:`

File: tox.ini (hunk 1 above)
Comment (pc-m) [reply tox.ini]: done

File: tox.ini (hunk 3 above)
Comment (fblackburn1) [reply tox.ini]: It is possible to make the inverse: tell to black to ignore E203?

**Comment by pc-m** [comment]
recheck

**Comment by sileht** [comment]
recheck

**Comment by pc-m** [comment]
recheck

**Comment by pc-m** [comment]
recheck

**Comment by pc-m** [comment]
recheck

**Comment by fredericlepied** [comment]
recheck

**Comment by pc-m** [comment]
recheck

**Comment by pc-m** [comment]
> * [dird-tox-integration ](https://zuul.wazo.community/logs/8/8/7fec05dd3a46939717bd3b20551f0084fcae081e/check/dird-tox-integration/cf677c8/) : FAILURE in 1h 34m 38s

this failure would be fixed with `export INTEGRATION_TEST_TIMEOUT=60`

**Pull request merged**